packages/keyring-eth-qr/src/device.ts

```diff
@@ -341,15 +341,14 @@ export class Device {
         ? DataType.transaction
         : DataType.typedTransaction;
 
-    const messageToSign = Buffer.from(
-      transaction.type === TransactionType.Legacy
-        ? RLP.encode(transaction.getMessageToSign())
-        : (transaction as FeeMarketEIP1559Transaction).getMessageToSign(),
+    const messageToSign = transaction.getMessageToSign();
+    const rawTransaction = Buffer.from(
+      Array.isArray(messageToSign) ? RLP.encode(messageToSign) : messageToSign,
```

Comment:
I initially kept this equal to how it is done in the original Keystone package, though we encode the transaction in the same way it is done in the Ledger keyring package.